skycatalogs/objects/snana_object.py

```diff
@@ -0,0 +1,205 @@
+import galsim
+import h5py
+import numpy as np
+from .base_object import BaseObject,ObjectCollection
+
+__all__ = ['SnanaObject', 'SnanaCollection']
+
+class SnanaObject(BaseObject):
+    _type_name = 'snana'
+
+    def __init__(self, ra, dec, id, object_type, belongs_to, belongs_index,
+                 redshift=None):
+        super().__init__(ra, dec, id, object_type, belongs_to, belongs_index,
+                         redshift)
+        self._mjds = None
+        self._lambda = None
+
+        # indices of elements in mjd array bounding our mjd
+        self._mjd_ix_l = None
+        self._mjd_ix_u = None
+
+    def _get_sed(self, mjd=None):
+        if mjd is None:
+            mjd = self._belongs_to._mjd
+        mjd_start = self.get_native_attribute('start_mjd')
+        mjd_end = self.get_native_attribute('end_mjd')
+        if mjd < mjd_start or mjd > mjd_end:
+            return None, 0.0
+
+        return self._linear_interp_SED(mjd), 0.0
+
+    def get_gsobject_components(self, gsparams=None, rng=None):
+        if gsparams is not None:
+            gsparams = galsim.GSParams(**gsparams)
+        return {'this_object': galsim.DeltaFunction(gsparams=gsparams)}
+
+    def get_observer_sed_component(self, component, mjd=None):
+        sed, _ = self._get_sed(mjd=mjd)
+        if sed is not None:
+            sed = self._apply_component_extinction(sed)
+        return sed
+
+    def get_LSST_flux(self, bandpass, sed=None, cache=True, mjd=None):
+        def _to_flux(m):
+            return np.exp(-0.921034037196184 * m)
+        def _to_mag(f):
+            return np.log(flux)/(-0.921034037196184)
+
+        flux = super().get_LSST_flux(bandpass, sed, mjd)
+
+        mjd_ix_l, mjd_ix_u, mjd_fraction = self._find_mjd_interval(mjd)
+        if flux > 0.0:
+            with h5py.File(self._belongs_to._SED_file, 'r') as f:
+                cors = f[self._id][f'magcor_{bandpass}']
```

Comment:
I agree it would be good to check.  I was going by the documentation Rob provided:

> magcor_<band>: multiple datasets, an ( nmjd ) array for each
     photometric band.  Has the magnitude correction that should be
     applied for this band to any synthetic photometry calculated from
     any processing of the SED.  This field is redundant with the
     previous two files, as magcor_<band> = mag_<band> - synmag_<band>

Reply:
I started by doing the calculation in magnitudes, then switched to fluxes since it involved less work.  I verified that the results were the same both ways.

Reply:
I've added some CI tests (skipped for now in the CI environment since data are not available) but did not add one for this because the required information is not currently available through the API.  However I did look directly at one of the hdf5 files and verified for one source, one band that (using names as they appear in the file) ```mag_<band> == synmag_<band> + magcor_<band>``` 
The first term is the result of using 5 A binning in the SED and the second is with 100 A binning.